src/authorization/authorization.test.ts

```diff
@@ -249,6 +249,69 @@ describe('API Key Interceptors', () => {
       });
       expect(response.config.params.email).toBe('helloworld@gmail.com');
     });
+
+    it('should try to generate JWT again with user ID if updateEmail is called after setUserID', async () => {
+      mockRequest.onPost('/users/updateEmail').reply(200, {
+        code: 'hello'
+      });
+
+      const mockGenerateJWT = jest
+        .fn()
+        .mockReturnValue(Promise.resolve(MOCK_JWT_KEY));
+      const { setUserID } = initialize('123', mockGenerateJWT);
+
+      await setUserID('mock-id');
+      await updateUserEmail('helloworld@gmail.com');
+      expect(mockGenerateJWT).toHaveBeenCalledTimes(2);
+      expect(mockGenerateJWT).toHaveBeenCalledWith({ userID: 'mock-id' });
+      expect(mockGenerateJWT).not.toHaveBeenCalledWith({
+        email: 'helloworld@gmail.com'
+      });
+
+      jest.advanceTimersByTime(60000 * 4.1);
```

Comment:
JWT tokens gets automatically refreshed one minute before expiry in the SDK code, so we also want to sanity check that after a refresh, we're still signing the JWT with the user ID and not the new email address.

This specific JWT is hard-coded to expire after 5 seconds, which is why we're jumping forward in time 4.1 seconds.